octodns/processor/arpa.py

```diff
@@ -37,7 +39,10 @@ def process_source_zone(self, desired, sources):
 
                 for ip in ips:
                     ptr = ip_address(ip).reverse_pointer
-                    self._records[f'{ptr}.'].add(record.fqdn)
+                    auto_arpa_priority = record.octodns.get('auto_arpa_priority', 999)
+                    self._records[f'{ptr}.'].append((auto_arpa_priority, record.fqdn))
+                    unique_list = list(set(self._records[f'{ptr}.']))
```

Comment:
I believe this will only keep things unique when they have the same priority since tuples of (priority, fqdn) are being placed into the list. so if the same fqdn showed up 

It may be preferable to just `append` everything here and then deal with making things unique down below when it's pulling the fqdn's out of the tuples. That'd probably result in a slightly more predictable behavior where the lowest priority seen for a fqdn would win.

Another option that comes to mind would be a `dict` with the fqdn a the key and priority as the value. That would result in last one winning though which is at least consistent, but probably harder to predict. Sorting it would require a couple of extra steps, something like `.items()`, flipping the k & v, and then sorting.

Reply:
Yes, I think you are right. This will only keep things unique if they have the same priority. I have moved it to populate instead.